wifimgr.py

```diff
@@ -186,12 +197,11 @@ def handle_configure(client, request):
         return False
     # version 1.9 compatibility
     try:
-        ssid = match.group(1).decode("utf-8").replace("%3F", "?").replace("%21", "!")
-        password = match.group(2).decode("utf-8").replace("%3F", "?").replace("%21", "!")
-    except Exception:
-        ssid = match.group(1).replace("%3F", "?").replace("%21", "!")
-        password = match.group(2).replace("%3F", "?").replace("%21", "!")
-
+        ssid = _unquote(match.group(1).decode("utf-8").replace("%3F", "?").replace("%21", "!"))
+        password = _unquote(match.group(2).decode("utf-8").replace("%3F", "?").replace("%21", "!"))
```

Comment:
why did you keep the replace for the %XX stuff here, isn't that something your new _unquote call should do?